src/lib/DpApi.js

```diff
@@ -39,28 +39,34 @@ const getHeaders = function ({ headers, url }) {
   }
 }
 
+const appendSerializedUrlParams = (url, params) => {
+  if (!params || Object.keys(params).length === 0) {
+    return url
+  }
+  params = stringify(params, { encodeValuesOnly: true, arrayFormat: 'brackets' })
+
+  // Url params may be already appended before passing it to dpApi, this must be handled accordingly.
+  return url.includes('?') ? `${url}&${params}` : `${url}?${params}`
+}
+
 const doRequest = (async ({ url, method = 'GET', data = {}, params, options = {} }) => {
-  const payload = {
-    method,
-    options,
-    params,
-    headers: getHeaders({ ...params, url })
+  const fetchOptions = {
+    headers: getHeaders({ ...params, url }),
```

Comment:
Its confusing.
Shouldn't this be something like
```suggestion
    headers: getHeaders({ headers, url }),
```
and we have to  pass `headers` as another param to the method?
I know it was like that before. I just wonder how it ever worked. :see_no_evil:

Reply:
it is a little bit confusing but what actually happens here is that all properties of `headers` are being spread into the object that is passed to `getHeaders`. otherwise you'd have to do something like

```
const { headers } = params
// or: const headers = params?.headers
headers: getHeaders({ headers, url })
```

and within `getHeaders` the object is destructured again, defining only the needed variables:

```
const getHeaders = function ({ headers, url }) {
```

Reply:
But If You have a look where we pass Headers, they are not within params, but next to them ;-) 
like so:
```
      return dpApi({
        method: 'PATCH',
        url: 'someUrl',
        headers: {
          'Content-type': 'application/vnd.api+json',
          Accept: 'application/vnd.api+json'
        }
```

Reply:
> I just wonder how it ever worked. 🙈

+1. i think that it did not, after dropping axios. It looks like the `headers` property in your last example (when using `dpApi` directly without the method aliases) was just ignored, as it is not forwarded to doRequest. as the api2defaultHeaders contain the correct headers anyways, this is just crucial for "a handful of 1.0 routes" that set custom headers.